Sources/ProjectDescription/ProjectOption.swift

```diff
@@ -3,35 +3,69 @@ import Foundation
 /// Additional options related to the `Project`
 public enum ProjectOption: Codable, Equatable {
     /// Text settings to override user ones for current project
-    case textSettings(TextSettings)
+    ///
+    /// - Parameters:
+    ///   - usesTabs: Use tabs over spaces.
+    ///   - indentWidth: Indent width.
+    ///   - tabWidth: Tab width.
+    ///   - wrapsLines: Wrap lines.
+    case textSettings(
+        usesTabs: Bool? = nil,
+        indentWidth: UInt? = nil,
+        tabWidth: UInt? = nil,
+        wrapsLines: Bool? = nil
+    )
 }
 
 // MARK: - Options + Codable
 
 extension ProjectOption {
-    enum CodingKeys: String, CodingKey {
+    private enum OptionsCodingKeys: String, CodingKey {
         case textSettings
     }
+    
+    private enum TextSettingsKeys: String, CodingKey {
+        case usesTabs
+        case indentWidth
+        case tabWidth
+        case wrapsLines
+    }
 
     public init(from decoder: Decoder) throws {
-        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let container = try decoder.container(keyedBy: OptionsCodingKeys.self)
 
         if container.allKeys.contains(.textSettings), try container.decodeNil(forKey: .textSettings) == false {
-            var associatedValues = try container.nestedUnkeyedContainer(forKey: .textSettings)
-            let textSettings = try associatedValues.decode(TextSettings.self)
-            self = .textSettings(textSettings)
+            let textSettingsContainer = try container.nestedContainer(
+                keyedBy: TextSettingsKeys.self,
+                forKey: .textSettings
+            )
+            
+            let usesTabs = try textSettingsContainer.decodeIfPresent(Bool.self, forKey: .usesTabs)
+            let indentWidth = try textSettingsContainer.decodeIfPresent(UInt.self, forKey: .indentWidth)
+            let tabWidth = try textSettingsContainer.decodeIfPresent(UInt.self, forKey: .tabWidth)
+            let wrapsLines = try textSettingsContainer.decodeIfPresent(Bool.self, forKey: .wrapsLines)
+            
+            self = .textSettings(
+                usesTabs: usesTabs,
+                indentWidth: indentWidth,
+                tabWidth: tabWidth,
+                wrapsLines: wrapsLines
+            )
```

Comment:
I would simplify this to
```suggestion
            let textSettingsContainer = try container.nestedContainer(keyedBy: TextSettingsKeys.self, forKey: .textSettings)
            self = .textSettings(
                usesTabs: try textSettingsContainer.decodeIfPresent(Bool.self, forKey: .usesTabs),
                indentWidth: try textSettingsContainer.decodeIfPresent(UInt.self, forKey: .indentWidth),
                tabWidth: try textSettingsContainer.decodeIfPresent(UInt.self, forKey: .tabWidth),
                wrapsLines: try textSettingsContainer.decodeIfPresent(Bool.self, forKey: .wrapsLines)
            )
```

Reply:
makes sense